src/libDirectoryService/DirectoryService.cpp

```diff
@@ -232,14 +232,14 @@ bool DirectoryService::ProcessSetPrimary(const vector<unsigned char>& message,
                                           sizeof(uint32_t));
         curr_offset += sizeof(uint32_t);
 
-        for (unsigned int i = 0; i < ds.size(); i++)
+        for (auto& d : ds)
```

Comment:
I wonder why this wasn't changed to `const`

Reply:
I guess it's because our `Serialize` was not declared as const